core/ir/synthetic/encoding_common.h

```diff
@@ -66,31 +71,41 @@
  * 2 bits, eflags: most significant bit set to 1 indicates the instruction reads at least
  * one arithmetic flag, least significant bit set to 1 indicates the instruction writes
  * at least one arithmetic flag.
- * 4 bits, #src: number of source operands (read) that are registers.
+ * 4 bits, #src: number of source operands (read) that are registers. Registers used in
+ * memory reference operands of the instruction we are encoding (regardless on whether
+ * they are source or destination operands) are considered as source operands in the
+ * encoded instruction.
  * 4 bits, #dst: number of destination operands (written) that are registers.
- * (Note that we are only interested in register dependencies, hence operands that are
- * not registers, such as immediates or memory references, are not present.)
  *
- * Following the 4 instruction-related bytes are the bytes for encoding register operands.
- * Each operand is 1 byte.
+ * We assume these encoded values to be little-endian.
+ * Note that we are only interested in register dependencies, hence operands that are
+ * not registers, such as immediates or memory references, are not present.
+ *
+ * Following the 4 header bytes are the bytes for encoding register operands.
+ * Each operand is 2 bytes.
+ * The first byte encodes the register id (#reg_id_t).
+ * The second byte encodes the register size in bytes following #opnd_size_t
+ * (e.g., OPSZ_4 = 4 bytes = 32 bits register).
```

Comment:
We had agreed to have a single operation size for the whole instruction, not per-operand.  (Having the size is not very different from not collapsing the register in the first place.)

Reply:
Fixed.